packages/migrations/src/batched-migrations/batched-migration-job.ts

```diff
@@ -0,0 +1,18 @@
+import { z } from 'zod';
+
+export const BatchedMigrationJobStatusSchema = z.enum(['pending', 'failed', 'succeeded']);
+export type BatchedMigrationJobStatus = z.infer<typeof BatchedMigrationJobStatusSchema>;
+
+export const BatchedMigrationJobRowSchema = z.object({
+  id: z.string(),
+  batched_migration_id: z.string(),
+  min_value: z.bigint({ coerce: true }),
+  max_value: z.bigint({ coerce: true }),
```

Comment:
I assume you are using `Bigint` to allow us to calculate with values from postgres BIGINT `id` columns? I’ve been wondering what to do with this because of the `table_name_to_jsonb()` sprocs we have in PT. Ideally I’d like to remove these sprocs and use `to_jsonb()` instead. The main issue will be that `to_jsonb()` translates postgres BIGINT into a JSON `Number`, so we can’t safely handle `id` columns larger than 2^53 (the biggest integer you can store in a JS `Number`). I was thinking that this might be fine in practice, because that feels like a safe number of rows to me. What do you think about just mapping postgres BIGINT to JS `Number` and not worrying about overflow?

Reply:
I've been thinking about adding the equivalent of PT's `db-types.ts` file to PL to give us zod and TS types that exactly (as close as possible) mirror the DB tables. I've found this to be super helpful in converting everything from `query()` to `queryValidated()` (as well as adding strict typing everywhere). Part of this requires deciding how to get data out of SQL queries, especially for nested data structures. For PT we've been using the `table_name_to_jsonb` sprocs, but this ties us into a long-term requirement for using sprocs. It would be much nicer to use `to_jsonb()` for this. The main blockers in doing this in PT right now are:
* treatment of `id` columns, where we currently export the BIGINTs as `text` to map to `z.string()`
* conversion of INTERVALs, where we currently convert them to milliseconds to map to `z.number()`

For intervals, `to_jsonb()` will convert to a string like `"00:32:20.32"` (HH:MM:SS.ssss). I was thinking to use a custom zod schema to parse this into a numeric value, like we do with our custom `DateFromISOString` schema for dates.

So that really only leaves `id` columns (and BIGINTs in general). The problem here is that `to_jsonb()` _doesn't_ convert these to strings so we can't use a custom zod schema to transform them to JavaScript `Bigint`. Instead, `to_jsonb()` converts them to JSON numbers, which is where we get the 2^53 restriction from.

This is all a long background to say that I think we should start with right now with a `db-types.ts` (or `.js`) file in PL containing a `BatchedMigrationJobCodec` zod type (except swap `Codec` for whatever we are going to use in PL) and this should have `min_value: z.number()` (not `z.bigint()`).

Reply:
And yes, this is for consistency, so that we can know that we can always use `to_jsonb()` in SQL to construct nested data and this will correctly validate with zod on the other side.

Reply:
Although, now I remembered that this is actually in a package, not PL, so it would actually be fine to use `z.bigint()` here. Sorry for all the noise!